akka-projection-slick/src/main/scala/akka/projection/slick/SlickProjection.scala

```diff
@@ -49,16 +58,58 @@ object SlickProjection {
       projectionId: ProjectionId,
       sourceProvider: SourceProvider[Offset, Envelope],
       databaseConfig: DatabaseConfig[P],
-      handler: SlickHandler[Envelope]): ExactlyOnceSlickProjection[Envelope] =
+      handler: SlickHandler[Envelope])(
+      implicit system: ActorSystem[_]): ExactlyOnceSlickProjection[Offset, Envelope] = {
+
+    // lift to Future handler
+    val liftedHandler =
+      new Handler[Envelope] {
+
+        val offsetStore: SlickOffsetStore[P] =
+          new SlickOffsetStore(databaseConfig.db, databaseConfig.profile, SlickSettings(system))
```

Comment:
I will move that to a constructor param of `SlickProjectionImpl` to avoid creating if in different places.